src/praisonai-bot/praisonai_bot/_guarded_subprocess.py

```
@@ -0,0 +1,174 @@
+    resolved_timeout = _default_timeout() if timeout is None else timeout
+    kwargs.setdefault("capture_output", True)
+    kwargs.setdefault("text", True)
+
+    # stdin at /dev/null turns "wait forever for input" into an immediate EOF
+    # for tools that honour it, so the timeout is a backstop rather than the
+    # primary defence.
+    if not interactive:
+        kwargs.setdefault("stdin", subprocess.DEVNULL)
+
+    try:
+        return subprocess.run(
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)